mk/aosp_optee.mk

```diff
@@ -19,42 +19,57 @@
 ##########################################################
 ## define common variables, like TA_DEV_KIT_DIR         ##
 ##########################################################
-OPTEE_OUT_DIR ?= $(realpath $(PRODUCT_OUT))/optee
+TOP_ROOT_ABS := $(realpath $(TOP))
+PREBUILT_MAKE ?= prebuilts/build-tools/linux-x86/bin/make
+ifneq (,$(wildcard $(PREBUILT_MAKE)))
+HOST_MAKE := $(PREBUILT_MAKE)
+ifneq (,$(filter -j%, $(MAKE)))
+HOST_MAKE += $(filter -j%, $(MAKE))
+endif
```

Comment:
Why is this needed? `$(MAKE)` should not contain flags as far as I can tell.

Reply:
hmm, when I use $(info) to print during aosp build, the value is something like "make -j20".
If we do not specify the -j value, then it seems to run make with -j1, which would take more time to finish the compiling, so here I add the -j option if it is specified in the value of $(MAKE)

Reply:
The `j` flag should normally be handled automatically by make. Otherwise you will not get the correct number of parallel processes. If you pass `-j 20` to sub-make, then it will be allowed to spawn 20 children, not counting processes running make targets in the parent makefile, so more than 20 actually.

You say make runs with `-j1` if you do not specify a value. I think it is because you changed `$(MAKE)` to `$(HOST_MAKE)`. The make programs does very specific things when is sees that $(MAKE) is invoked in a recipe, and honestly I think we should not change that.

Please try this instead:
```
override MAKE = $(PREBUILT_MAKE)
```

(and disregard my comment about adding `+` since it is not needed in this case).

Reply:
hmm, for the -j option, the value is half of the even number I specified when I run make droidcore, like when I run make -j40 droidcore from the root of aosp,  then I would get "make -j20" when I use $(info) to print $(MAKE) in the  mk/aosp_optee.mk, if I run "make -j8 droidcore" then I would get "make -j4" for $(MAKE), I am not sure who sets the -j option for the $(MAKE) variable, 
so here I just get the -j value from $(MAKE) and use that with the prebuilt make command.

Reply:
if I only do like this:
PREBUILT_MAKE ?= prebuilts/build-tools/linux-x86/bin/make
MAKE := $(PREBUILT_MAKE)

then what I get for $(MAKE) is only "prebuilts/build-tools/linux-x86/bin/make", there is not any option specified explicitly.

Reply:
Yes, sounds good, do things work as expected then?

Reply:
still not quite sure about the $MAKE, but the MAKE variable looks like defined here:
https://android.googlesource.com/platform/build/kati/+/master/main.cc#100

Reply:
> Yes, sounds good, do things work as expected then?

from my local test, it seem overriding MAKE would make the -j option lost, and cause other places that use MAKE to run with less threads.
I think we'd better not to override MAKE variant

Reply:
@jforissier OK, I updated to set MAKE for the targets only, please help to check again to see if you are OK with the changes.

Thanks,
Yongqin Liu

Reply:
@liuyq I'd prefer that, but will it run parallel builds?

Considering https://android.googlesource.com/platform/build/kati/+/master/main.cc#100 again, I now have a doubt. What may be happening is, perhaps the AOSP top-level build tool (kati) does not keep track of the number of make sub-processes, contrary to the real make. Therefore the redefinition of `$(MAKE)` to include `-jx` would be a dirty hack to still enable parallel make. A hack that we would therefore need to preserve :(

If my analysis is correct, then your initial proposal is OK, it just needs a comment stating that the AOSP build tool is not the regular make, that it adds `-jN` to `$(MAKE)`, and that we should preserve the flag or we would lose parallel build.

Reply:
> @liuyq I'd prefer that, but will it run parallel builds?
I don't think so, by default make would run with only 1 thread when no -j option specified I guess.
 
> If my analysis is correct, then your initial proposal is OK, it just needs a comment stating that the AOSP build tool is not the regular make, that it adds `-jN` to `$(MAKE)`, and that we should preserve the flag or we would lose parallel build.

Let's use my initial proposal with comments inline, and btw @vchong reminders me that we could not set MAKE to the prebuilt make directly, since the P build does not have the prebuilt make by default, so we need to do some check there too.

Reply:
Thanks @liuyq, that looks good to me. You may add my `Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>` to the second patch.

One minor thing: the checkpatch warning. You can get rid of it by putting long URLs in a Link: tag, like this:

```
mk/aosp_optee.mk: use prebuilt make command

The host side make command is disallowed by the Android build system on
AOSP master [1]. Using it anyway causes the following build error:

  "make" is not allowed to be used.

See [2] for more information.

Link: [1] https://android-review.googlesource.com/c/platform/build/soong/+/796565
Link: [2] https://android.googlesource.com/platform/build/+/master/Changes.md#PATH_Tools
Signed-off-by: Yongqin Liu <yongqin.liu@linaro.org>
```

Reply:
thanks for the commit message, will update as you suggested here.